dual_net.py

```diff
@@ -301,61 +312,39 @@ def logging_ops():
         name="weight_summaries")
 
 
+def compute_gradient_ratio(weight_tensors, before_weights, after_weights):
```

Comment:
This actually doesn't compute anything about the gradient, this is computing  || delta w || / || w|| for all weight tensors w. The gradient is related but some optimizers have complicated weight updates. I would call this function compute_update_ratio instead.